spec/reek/smell_detectors/module_initialize_spec.rb

```diff
@@ -86,4 +86,44 @@ def initialize; end
 
     expect(src).not_to reek_of(:ModuleInitialize)
   end
+
+  it 'reports nothing for a method named initialize in a class assigned to a let variable' do
+    src = <<-RUBY
+      module Alfa
+        RSpec.describe Bravo do
+          let(:klass) do
+            Class.new do
+              def initialize; end
+            end
+          end
+        end
+      end
+    RUBY
+
+    expect(src).not_to reek_of(:ModuleInitialize)
+  end
+
+  it 'reports nothing for a DSL-based' do
```

Comment:
Seems like that description is incomplete?

Reply:
What sounds correct to you?

```ruby
  it 'reports nothing for a DSL-based method' do
```
```ruby
  it 'reports nothing for a DSL-based method initializing a class' do
```